src/dodal/devices/undulator.py

```diff
@@ -24,11 +24,36 @@ def __init__(
         self,
         prefix: str,
         name: str = "",
+        poles: int | None = None,
+        length: float | None = None,
     ) -> None:
+        """Constructor
+
+        Args:
+            prefix: PV prefix
+            poles (int): Number of magnetic poles built into the undulator
+            length (float): Length of the undulator in meters
+            name (str, optional): Name for device. Defaults to "".
+        """
+
         with self.add_children_as_readables():
             self.gap_motor = Motor(prefix + "BLGAPMTR")
             self.current_gap = epics_signal_r(float, prefix + "CURRGAPD")
             self.gap_access = epics_signal_r(UndulatorGapAccess, prefix + "IDBLENA")
-        self.gap_discrepancy_tolerance_mm: float = UNDULATOR_DISCREPANCY_THRESHOLD_MM
+
+        with self.add_children_as_readables(ConfigSignal):
+            self.gap_discrepancy_tolerance_mm, _ = soft_signal_r_and_backend(
+                float,
+                UNDULATOR_DISCREPANCY_THRESHOLD_MM,
+            )
+            if poles is not None:
+                self.poles, _ = soft_signal_r_and_backend(int, poles)
+            else:
+                self.poles = None
+
+            if length is not None:
+                self.length, _ = soft_signal_r_and_backend(float, length)
+            else:
+                self.length = None
```

Comment:
For some reason, the initial values are not propagating to the signal backends, TODO: investigate

Reply:
Resolved by #525